### ISSUE TITLE
Crash while reading locked .editorconfig file

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#15096]
Seen in an integration test run:

```
The process cannot access the file 'd:\j\workspace\windows_vsi_p---ad02e19d\Open\Binaries\Temp\0peyk1s2.zh0\CSProj\.editorconfig' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) in f:\dd\ndp\clr\src\BCL\system\io\__error.cs:line 184
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) in f:\dd\ndp\clr\src\BCL\system\io\filestream.cs:line 878
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy) in f:\dd\ndp\clr\src\BCL\system\io\filestream.cs:line 595
   at System.IO.FileStream..ctor(String path, FileMode mode) in f:\dd\ndp\clr\src\BCL\system\io\filestream.cs:line 512
   at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfig.<ParseAsync>d__11.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) in f:\dd\ndp\clr\src\BCL\system\runtime\compilerservices\TaskAwaiter.cs:line 176
   at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache.<CacheDirectoryAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) in f:\dd\ndp\clr\src\BCL\system\runtime\compilerservices\TaskAwaiter.cs:line 176
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in f:\dd\ndp\clr\src\BCL\system\runtime\compilerservices\TaskAwaiter.cs:line 156
   at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache.<GetConfigsAsync>d__1.MoveNext()
```

**Comment by Pilchie** [comment]
@jasonmalinowski, @jmarolf, @olegtk - where should the `catch` happen here?  In ` Microsoft.VisualStudio.CodingConventions`, or in Roslyn?

**Comment by olegtk** [comment]
Hmmm, Microsoft.VisualStudio.CodingConventions should probably use FileShare.Read when reading the file, that should prevent this exception.

**Comment by Pilchie** [comment]
That would be good, but probably not enough.  IO can *always* fail, and the code should be prepared to handle that without crashing.  My question is which layer should be handling it?

**Comment by olegtk** [comment]
You are right. The lib cannot really handle IO exceptions in a meaningful way so I'd suggest this should be handled on Roslyn side, where you have at least more options to handle it.

**Comment by jasonmalinowski** [comment]
We can handle the exception, but @olegtk still do FileShare.Read to play nice(r) with others.

**Comment by olegtk** [comment]
Sure. Can you please file an internal bug for this? Thanks!

**Comment by Pilchie** [comment]
Filed internal bug 292101 for this.

**Comment by Pilchie** [comment]
I think we still need to catch in the Roslyn code too.

**Comment by jmarolf** [comment]
fixed with PR #15669
